account_payment_other_company/models/account_journal.py

```diff
@@ -36,3 +36,51 @@ def _name_search(self, name, args=None, operator='ilike',
             return super()._name_search(name, args,
                                         operator, limit,
                                         name_get_uid)
+
+    @api.model
+    def search_read(self, domain=None, fields=None, offset=0,
+                    limit=None, order=None):
+        if self.env.context.get('sudo', False):
+            records = self.sudo().\
+                search(domain or [], offset=offset, limit=limit, order=order)
+            if not records:
+                return []
+
+            if fields and fields == ['id']:
+                # shortcut read if we only want the ids
+                return [{'id': record.id} for record in records]
+            if 'active_test' in self._context:
+                context = dict(self._context)
+                del context['active_test']
+                records = records.with_context(context)
+
+            result = records.read(fields)
+            if len(result) <= 1:
+                return result
+
+            # reorder read
+            index = {vals['id']: vals for vals in result}
+            return [index[record.id] for
+                    record in records if record.id in index]
+        else:
+            records = self.\
+                search(domain or [], offset=offset, limit=limit, order=order)
+            if not records:
+                return []
+
+            if fields and fields == ['id']:
+                # shortcut read if we only want the ids
+                return [{'id': record.id} for record in records]
+            if 'active_test' in self._context:
+                context = dict(self._context)
+                del context['active_test']
+                records = records.with_context(context)
+
+            result = records.read(fields)
+            if len(result) <= 1:
+                return result
+
+            # reorder read
+            index = {vals['id']: vals for vals in result}
+            return [index[record.id] for
+                    record in records if record.id in index]
```

Comment:
@osi-scampbell What about this?

```suggestion
            records = self.sudo().\
                search(domain or [], offset=offset, limit=limit, order=order)
        else:
            records = self.\
                search(domain or [], offset=offset, limit=limit, order=order)
       
            if not records:
                return []

            if fields and fields == ['id']:
                # shortcut read if we only want the ids
                return [{'id': record.id} for record in records]
            if 'active_test' in self._context:
                context = dict(self._context)
                del context['active_test']
                records = records.with_context(context)

            result = records.read(fields)
            if len(result) <= 1:
                return result

            # reorder read
            index = {vals['id']: vals for vals in result}
            return [index[record.id] for
                    record in records if record.id in index]
```

Also there is no call to `super()`.